src/server/command_registry.cc

```diff
@@ -191,35 +204,66 @@ CommandRegistry& CommandRegistry::operator<<(CommandId cmd) {
 
   absl::InlinedVector<std::string_view, 2> maybe_subcommand = StrSplit(cmd.name(), " ");
   const bool is_sub_command = maybe_subcommand.size() == 2;
-  if (const auto it = cmd_rename_map_.find(maybe_subcommand.front()); it != cmd_rename_map_.end()) {
+  auto it = cmd_rename_map_.find(maybe_subcommand.front());
+  if (it != cmd_rename_map_.end()) {
     if (it->second.empty()) {
       return *this;  // Incase of empty string we want to remove the command from registry.
     }
     k = is_sub_command ? absl::StrCat(it->second, " ", maybe_subcommand[1]) : it->second;
   }
 
+  std::optional<CommandId> cloned = std::nullopt;
+  for (const auto& [alias, target] : alias_map_) {
+    if (target == k) {
+      cloned.emplace(cmd.Clone(alias));
+      alias_map_.erase(alias);
+      break;
+    }
+  }
+
   if (restricted_cmds_.find(k) != restricted_cmds_.end()) {
     cmd.SetRestricted(true);
+    if (cloned.has_value()) {
+      cloned->SetRestricted(true);
+    }
   }
 
   if (oomdeny_cmds_.find(k) != oomdeny_cmds_.end()) {
     cmd.SetFlag(CO::DENYOOM);
+    if (cloned.has_value()) {
+      cloned->SetFlag(CO::DENYOOM);
+    }
   }
 
   cmd.SetFamily(family_of_commands_.size() - 1);
-  if (acl_category_)
+  if (cloned.has_value()) {
+    cloned->SetFamily(cmd.GetFamily());
+  }
+  if (acl_category_) {
     cmd.SetAclCategory(*acl_category_);
+    if (cloned.has_value()) {
+      cloned->SetAclCategory(*acl_category_);
+    }
+  }
 
   if (!is_sub_command || absl::StartsWith(cmd.name(), "ACL")) {
     cmd.SetBitIndex(1ULL << bit_index_);
-    family_of_commands_.back().push_back(std::string(k));
+    family_of_commands_.back().emplace_back(k);
     ++bit_index_;
+
+    if (cloned) {
+      cloned->SetBitIndex(1ULL << bit_index_);
+      family_of_commands_.back().emplace_back(cloned->name());
+      ++bit_index_;
+    }
   } else {
     DCHECK(absl::StartsWith(k, family_of_commands_.back().back()));
     cmd.SetBitIndex(1ULL << (bit_index_ - 1));
   }
```

Comment:
This code path does not seem to be taken by any commands in the tree afaics (is sub command as is not acl)